src/zarr/testing/store.py

```diff
@@ -319,25 +330,43 @@ async def test_set_if_not_exists(self, store: S) -> None:
         result = await store.get("k2", default_buffer_prototype())
         assert result == new
 
-    async def test_getsize(self, store: S) -> None:
-        key = "k"
-        data = self.buffer_cls.from_bytes(b"0" * 10)
-        await self.set(store, key, data)
-
-        result = await store.getsize(key)
-        assert isinstance(result, int)
-        assert result > 0
-
-    async def test_getsize_raises(self, store: S) -> None:
-        with pytest.raises(FileNotFoundError):
-            await store.getsize("not-a-real-key")
-
-    async def test_getsize_prefix(self, store: S) -> None:
-        prefix = "array/c/"
-        for i in range(10):
-            data = self.buffer_cls.from_bytes(b"0" * 10)
-            await self.set(store, f"{prefix}/{i}", data)
-
-        result = await store.getsize_prefix(prefix)
-        assert isinstance(result, int)
-        assert result > 0
+
+class LatencyStore(WrapperStore[Store]):
+    """
+    A wrapper class that takes any store class in its constructor and
+    adds latency to the `set` and `get` methods. This can be used for
+    performance testing.
+    """
+
+    get_latency: float
+    set_latency: float
+
+    def __init__(self, cls: Store, *, get_latency: float = 0, set_latency: float = 0) -> None:
+        self.get_latency = float(get_latency)
+        self.set_latency = float(set_latency)
+        self._store = cls
+
+    async def set(self, key: str, value: Buffer) -> None:
+        await asyncio.sleep(self.set_latency)
+        await self._store.set(key, value)
+
+    async def get(
+        self, key: str, prototype: BufferPrototype, byte_range: ByteRangeRequest | None = None
+    ) -> Buffer | None:
+        """
+        Add latency to the get method.
+
+        Adds a sleep of `self.get_latency` seconds before calling the wrapped method.
+
+        Parameters
+        ----------
+        key : str
+        prototype : BufferPrototype
+        byte_range : ByteRangeRequest, optional
+
+        Returns
+        -------
+        buffer : Buffer or None
+        """
```

Comment:
either remove this to use the base classes doc string or add a similar one to `set`

Reply:
i went for the latter